### PR TITLE
feat: add full support for aws access keys

### PR DESCRIPTION
**Pull request by yrobla**
Closes: #1669

### CODE REVIEW

File: src/modules/aws_iam/mappers/access_key.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  cloud = new Crud2<AccessKey>({
+    create: async (es: AccessKey[], ctx: Context) => {
+      // Do not cloud create, just restore database
+      await this.module.accessKey.db.delete(es, ctx);
+    },
+    read: async (ctx: Context, id?: string) => {
+      const client = (await ctx.getAwsClient()) as AWS;
+
+      if (id) {
+        const { keyId, user } = this.idFields(id);
```
Comment (aguillenv) [inline src/modules/aws_iam/mappers/access_key.ts:68]: I do not see `user` decorated with `@cloudId`, how is this working?

**Review by aguillenv — COMMENTED** [review]
the RPC index is not exporting the fn, also, can we add some tests for the RPC call?

File: src/modules/aws_iam/mappers/access_key.ts (hunk 1 above)
Comment (yrobla) [reply src/modules/aws_iam/mappers/access_key.ts:68]: yep, an accidental removal... i will readd

File: src/modules/aws_iam/entity/access_key.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    enum: accessKeyStatusEnum,
+  })
+  status?: accessKeyStatusEnum;
+
+  /**
+   * @public
+   * Reference to the user for this record
+   */
+  @ManyToOne(() => IamUser, {
+    cascade: true,
```
Comment (aguillenv) [inline src/modules/aws_iam/entity/access_key.ts:83]: also, cascade here would not trigger the creation of the user for every access_key created?

**Comment by yrobla** [comment]
> the RPC index is not exporting the fn, also, can we add some tests for the RPC call?

there are tests for the RPC call .. the access_key_request() ones

**Comment by aguillenv** [comment]
> > the RPC index is not exporting the fn, also, can we add some tests for the RPC call?
> 
> there are tests for the RPC call .. the access_key_request() ones

oh sorry, I missed it!

File: src/modules/aws_iam/entity/access_key.ts [hunk 3]
```diff
@@ -79,10 +79,7 @@ export class AccessKey {
    * @public
    * Reference to the user for this record
    */
-  @ManyToOne(() => IamUser, {
-    cascade: true,
-    eager: true,
```
Comment (aguillenv) [inline src/modules/aws_iam/entity/access_key.ts]: oh, why did you remove the eager too?

Comment (yrobla) [reply src/modules/aws_iam/entity/access_key.ts]: my bad, i thought it was not needed

**Review by aguillenv — APPROVED** [review]

**Review by dfellis — APPROVED** [review]
